### PR TITLE
Add support for OAuth with provisioner Client

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Release Note

```release-note

```

### CODE REVIEW

File: config/config-matterwick.default.json [hunk 1]
```diff
@@ -60,5 +60,10 @@
       "CWSPublicPort": "",
       "CWSPrivatePort": ""
       
+  },
+  "CloudAuth": {
+    "ClientID": "Matterwick",
+    "ClientSecret": "Matterwick",
+    "TokenEndpoint": "https://matterwick.auth.mattermost.com/oauth/token"
```
Comment (esarafianou) [inline config/config-matterwick.default.json:67]: Why do we need the dummy data and how do they get overwritten?

File: internal/cloudtools/installation.go [hunk 2]
```diff
@@ -8,12 +8,8 @@ import (
 // GetInstallationIDFromOwnerID returns the installation that matches a given
 // OwnerID. Multiple matches will return an error. No match will return
 // an empty ID and no error.
-func GetInstallationIDFromOwnerID(serverURL, awsAPIKey, ownerID string) (*cloud.InstallationDTO, error) {
-	headers := map[string]string{
-		"x-api-key": awsAPIKey,
-	}
-	cloudClient := cloud.NewClientWithHeaders(serverURL, headers)
-	installations, err := cloudClient.GetInstallations(&cloud.GetInstallationsRequest{
+func GetInstallationIDFromOwnerID(client *cloud.Client, serverURL, awsAPIKey, ownerID string) (*cloud.InstallationDTO, error) {
```
Comment (esarafianou) [inline internal/cloudtools/installation.go:11]: We don't seem to need the `awsAPIKey` anymore, right?

**Review by esarafianou — CHANGES_REQUESTED** [review]
2 minor comments

File: internal/cloudtools/installation.go (hunk 2 above)
Comment (nickmisasi) [reply internal/cloudtools/installation.go]: Yep, removed.

**Review by esarafianou — APPROVED** [review]

**Review by gabrieljackson — APPROVED** [review]
LGTM

**Pull request merged**